launch/client.py

```diff
@@ -260,7 +260,7 @@ def __init__(
         self.bundle_location_fn: Optional[Callable[[], str]] = None
         self.batch_csv_location_fn: Optional[Callable[[], str]] = None
         self.configuration = Configuration(
-            host=self.endpoint + SCALE_LAUNCH_V1_PATH,
+            host=self.endpoint + SCALE_LAUNCH_V1_PATH if endpoint is None else self.endpoint,
```

Comment:
this is breaking custom endpoints

Reply:
I think this is incompatible with fed though because they use custom scaleapi + custom launch cc: @edwardpark97

Reply:
@yunfeng-scale what's your use case? i don't quite understand how you're using this

for the federal use case, I was trying to point the API calls away from https://api.scale.com (which is default) to use https://api.scalegov.com or even http://localhost:3000 instead. because some of these API calls go to `/v1/hosted_inference` and some of these API calls go to `/v1/launch` instead, it seemed like this was the best way of doing so

happy to hear any solution that would fit both of our use cases

Reply:
we need to do some integration testing with private (auto?) deployments like `https://launch-botocore-sessio-9089008.ml-internal.scale.com` there's no extra routing for this URL. it sounds to me we would want to add a field to clarify if the supplied endpoint path is private

Reply:
for now can you keep it this way since we urgently want to deploy launch changes to prepare for tomorrow's event

Reply:
@yunfeng-scale is it possible to revert this commit now to support the federal use case that @edwardpark97 mentioned?

Reply:
reverting this is still going to break integration test.

i think we could add another flag support your use case. when endpoint is set and that flag is set, use `self.endpoint + SCALE_LAUNCH_V1_PATH` as host